group.c

```diff
@@ -0,0 +1,274 @@
+/*
+ * group.c : Functions handling group entries retrieval.
+ */
+
+#include "nss-rightscale.h"
+#include "utils.h"
+
+#include <errno.h>
+#include <grp.h>
+#include <malloc.h>
+#include <pwd.h>
+#include <string.h>
+#include <unistd.h>
+
+/* struct used to store data used by getgrent. */
+struct group_data {
+    int group_counter;
+    struct group** users;
```

Comment:
Oh no! The `*` placement debate! I'm squarely in the `struct group **users;` camp because I believe that the arguments for it are sound, but I don't know if we really care about it enough for our one C project.